### PR TITLE
Add EVP_chacha20_poly1305() to speed.cc

### PR DESCRIPTION
**Pull request by geedo0**
### Issues:
Extension of PR #1311

### Description of changes: 
EVP_chacha20_poly1305 is an alternative interface for ChaCha20-Poly1305.
Notably, it uses an implementation of the algorithm that is distinct
from the preferred AEAD interfaced. This implementation, while still
assembly optimized, is older and slower.

### Call-outs:
Point out areas that need special attention or support during the review process. Discuss architecture or design changes.

The new interface is 5x slower because of the slower poly1305 implementation.

### Testing:
How is this change tested (unit tests, fuzz tests, etc.)? Are there any testing steps to be verified by the reviewer?

Initial results when run on a Macbook rocking an Intel Core i7.

```
/Users/gcr/git-workplace/aws-lc/build/clion/tool/bssl speed -filter ChaCha20-Poly1305
Did 3067000 EVP-ChaCha20-Poly1305 encrypt init operations in 1000188us (3066423.5 ops/sec)
Did 929000 EVP-ChaCha20-Poly1305 encrypt (16 bytes) operations in 1000357us (928668.5 ops/sec): 14.9 MB/s
Did 595000 EVP-ChaCha20-Poly1305 encrypt (256 bytes) operations in 1000757us (594549.9 ops/sec): 152.2 MB/s
Did 203000 EVP-ChaCha20-Poly1305 encrypt (1350 bytes) operations in 1000075us (202984.8 ops/sec): 274.0 MB/s
Did 43000 EVP-ChaCha20-Poly1305 encrypt (8192 bytes) operations in 1016349us (42308.3 ops/sec): 346.6 MB/s
Did 22000 EVP-ChaCha20-Poly1305 encrypt (16384 bytes) operations in 1016540us (21642.0 ops/sec): 354.6 MB/s
Did 3070000 EVP-ChaCha20-Poly1305 decrypt init operations in 1000062us (3069809.7 ops/sec)
Did 913000 EVP-ChaCha20-Poly1305 decrypt (16 bytes) operations in 1000170us (912844.8 ops/sec): 14.6 MB/s
Did 592000 EVP-ChaCha20-Poly1305 decrypt (256 bytes) operations in 1001610us (591048.4 ops/sec): 151.3 MB/s
Did 204000 EVP-ChaCha20-Poly1305 decrypt (1350 bytes) operations in 1003892us (203209.1 ops/sec): 274.3 MB/s
Did 43000 EVP-ChaCha20-Poly1305 decrypt (8192 bytes) operations in 1020958us (42117.3 ops/sec): 345.0 MB/s
Did 22000 EVP-ChaCha20-Poly1305 decrypt (16384 bytes) operations in 1022408us (21517.8 ops/sec): 352.5 MB/s
Did 17964750 AEAD-ChaCha20-Poly1305 seal init operations in 1000010us (17964570.4 ops/sec)
Did 5193500 AEAD-ChaCha20-Poly1305 seal (16 bytes) operations in 1000034us (5193323.4 ops/sec): 83.1 MB/s
Did 3533250 AEAD-ChaCha20-Poly1305 seal (256 bytes) operations in 1000006us (3533228.8 ops/sec): 904.5 MB/s
Did 1219000 AEAD-ChaCha20-Poly1305 seal (1350 bytes) operations in 1000483us (1218411.5 ops/sec): 1644.9 MB/s
Did 270000 AEAD-ChaCha20-Poly1305 seal (8192 bytes) operations in 1001246us (269664.0 ops/sec): 2209.1 MB/s
Did 139000 AEAD-ChaCha20-Poly1305 seal (16384 bytes) operations in 1004801us (138335.8 ops/sec): 2266.5 MB/s

Process finished with exit code 0
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: tool/speed.cc [hunk 1]
```diff
@@ -2627,6 +2627,7 @@ bool Speed(const std::vector<std::string> &args) {
        !SpeedECMUL(selected) ||
        // OpenSSL 1.0 doesn't support Scrypt
        !SpeedScrypt(selected) ||
+       !SpeedAESGeneric(EVP_chacha20_poly1305(), "EVP-ChaCha20-Poly1305", kTLSADLen, selected) ||
```
Comment (nebeid) [inline tool/speed.cc:2630]: Should the function name be changed because it has AES in it? Or should ChaCha20-Poly go in its own function?

Comment (geedo0) [reply tool/speed.cc:2630]: Different function name is the way to go. There's nothing specific to AES in this function.

Comment (WillChilds-Klein) [reply tool/speed.cc:2630]: perhaps `SpeedAEADGeneric`?

**Review by WillChilds-Klein — COMMENTED** [review]
code changes LGTM. failing integration tests might be related to issues addressed by @samuel40791765 in [this PR](https://github.com/aws/aws-lc/pull/1341).

**Review by WillChilds-Klein — APPROVED** [review]

**Review by andrewhop — APPROVED** [review]

**Review by WillChilds-Klein — APPROVED** [review]

**Pull request merged**